### PR TITLE
Cscap 42 Fix CallResult format and add CallError as well; Refactoring included

### PR DESCRIPTION
**Pull request by Xerasin**
Handle CallResult properly, I was in properly handling everything as a just a Call. I looked closer at the documentation and noted that CallResult is different and does not contain the action name for the message. You are supposed to infer it from the a message we sent and match up the message IDs. I'm not sure I should be deleting the previous messages from hash table, it may be nice to keep it?

I also added the ability to handle CallError though I'm unsure if this is the right format as it's mostly unstructured. 

I also added a more robust way of handling OnOCPPMessage listeners, now you need to specify a class you wish to listen for and it allows more than one listener to be registered at once.

Call
image
CallResult
image2

I also did some major refactoring and moved some classes into packages because the package was getting congested.

IF MERGED: will probably break #16 since there's now a abstract method filled out by OCPPMessageRequest,  and OCPPMessageResponse. Classes should extend one of these instead of OCPPMessage.

### CODE REVIEW

**Review by pglob — APPROVED** [review]
Approved with comments

**Comment by cfarhan** [comment]
I have quite a lot of work done for my CSCAP-51 and CSCAP-53 tickets I'm still working on. This PR might prove to be pretty troublesome if it gets merged before my tickets. Could this wait until after my changes are merged? I'll try to get both merged by Wednesday this week. I also did a bit of code reorganization with packages. 
image

**Review by cfarhan — CHANGES_REQUESTED** [review]
Commenting to prevent this PR from being merged. There are several changes that would break my current tickets I'm working on that I'd like to get merged first before this. I think trying to rebase off master if this ticket got merged would be a total nightmare. Let's hold off on merging this for the time being if that works with everyone.

**Comment by pglob** [comment]
If this PR, `Cscap 42 Fix CallResult format and add CallError as well; Refactoring included`, extends out into next Sprint, that is okay; We should still be on schedule.

**Comment by Xerasin** [comment]
> Commenting to prevent this PR from being merged. There are several changes that would break my current tickets I'm working on that I'd like to get merged first before this. I think trying to rebase off master if this ticket got merged would be a total nightmare. Let's hold off on merging this for the time being if that works with everyone.

That was my plan!

File: backend/src/main/java/com/sim_backend/websockets/OCPPWebSocketClient.java
Comment (Xerasin) [reply backend/src/main/java/com/sim_backend/websockets/OCPPWebSocketClient.java:76]: I catch all of them now and print a message to the stderr instead.

**Review by cfarhan — APPROVED** [review]

**Pull request merged**
